### PR TITLE
[#176412088] Add io-functions-cgn

### PR DESCRIPTION
**Pull request by AleDore**
This PR is intended to add brand new io-functions-cgn functions to mock images.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ services:
       - generated-env/env.io-fixtures
     build:
       context: ${IO_FIXTURES_CONTEXT}
-      dockerfile: ./Dockerfile
+      dockerfile: io-fixtures/Dockerfile
```
Comment (BurnedMarshal) [inline docker-compose.yml:23]: The base directory should come from `IO_FIXTURES_CONTEXT` context

File: env.io-functions-cgn [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+COSMOSDB_CGN_DATABASE_NAME=${COSMOSDB_NAME}
+COSMOSDB_CONNECTION_STRING=AccountEndpoint=${COSMOSDB_URI};AccountKey=${COSMOSDB_KEY};
+
+# needed to connect to cosmosdb server
+NODE_TLS_REJECT_UNAUTHORIZED=0
+AZURE_ENABLE_STRICT_SSL=false
+
+SERVICES_API_URL=http://host.docker.internal:${API_GATEWAY_PORT}/functions-services
+
+#TEST_INPS_SERVICE_ENDPOINT=http://webhook.site/INPS-1-OK
```
Comment (BurnedMarshal) [inline env.io-functions-cgn:23]: Could be removed?

File: env.io-functions-cgn [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+TEST_FISCAL_CODES=AAAAAA00A00A000B,AAAAAA00A00A000G
+
+SLOT_TASK_HUBNAME=IoFunctionCgn
+#SERVICES_API_URL=https://api.io.italia.it
```
Comment (BurnedMarshal) [inline env.io-functions-cgn:27]: Can be removed safely

File: env.io-functions-cgn [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+COSMOSDB_CONNECTION_STRING=AccountEndpoint=${COSMOSDB_URI};AccountKey=${COSMOSDB_KEY};
+
+# needed to connect to cosmosdb server
+NODE_TLS_REJECT_UNAUTHORIZED=0
+AZURE_ENABLE_STRICT_SSL=false
+
+SERVICES_API_URL=http://host.docker.internal:${API_GATEWAY_PORT}/functions-services
+
+#TEST_INPS_SERVICE_ENDPOINT=http://webhook.site/INPS-1-OK
+TEST_FISCAL_CODES=AAAAAA00A00A000B,AAAAAA00A00A000G
```
Comment (BurnedMarshal) [inline env.io-functions-cgn:24]: How is used this inside `io-fn-cgn`?

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: env.io-functions-cgn (hunk 4 above)
Comment (AleDore) [reply env.io-functions-cgn]: I'll remove it

File: env.io-functions-cgn (hunk 3 above)
Comment (AleDore) [reply env.io-functions-cgn]: Removed :)

File: env.io-functions-cgn (hunk 2 above)
Comment (AleDore) [reply env.io-functions-cgn]: Yep, I'll remove it

**Review by BurnedMarshal — COMMENTED** [review]
If the `io-fn-cgn` is called from the `io-backend` maybe some ENVs must be added to the `io-backend` configuration.

**Comment by AleDore** [comment]
> If the `io-fn-cgn` is called from the `io-backend` maybe some ENVs must be added to the `io-backend` configuration.

Yes, I'm waiting to merge [this](https://github.com/pagopa/io-backend/pull/751) before applying changes on backend env :)

**Comment by AleDore** [comment]
Can we merge this @BurnedMarshal

**Comment by AleDore** [comment]
> Can we merge this @BurnedMarshal ?

File: io-functions-cgn/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,11 @@
+FROM functions
+
+ARG FUNCTIONS_BONUS_COMMIT_SHA
```
Comment (BurnedMarshal) [inline io-functions-cgn/Dockerfile:3]:
```suggestion
ARG FUNCTIONS_CGN_COMMIT_SHA
```

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

**Review by BurnedMarshal — APPROVED** [review]

**Pull request merged**